ethereum/contracts/ETHVault.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract ETHVault is Ownable {
+    // Accepts ETH from the caller.
+    function lock() 
```

Comment:
So the ether asset is a special case and we should ideally implement the `receive` function instead: https://docs.soliditylang.org/en/latest/contracts.html#receive-ether-function

Reply:
Also I think its fine for the anyone untrusted to deposit ether. In fact we need to allow that if there ever comes a time when we need to recover from a hack and add funds to compensate from a hack.

Reply:
and we should emit a `Deposit(account, amount)` too, where `account` is the address of the app.